certbot-postfix/certbot_postfix/installer.py

```diff
@@ -0,0 +1,289 @@
+            raise errors.NotSupportedError('Postfix version must be at least %s', version_string)
+
+    def _lock_config_dir(self):
+        """Stop two Postfix plugins from modifying the config at once.
+
+        :raises .PluginError: if unable to acquire the lock
+
+        """
+        try:
+            certbot_util.lock_dir_until_exit(self.conf('config-dir'))
+        except (OSError, errors.LockError):
+            logger.debug("Encountered error:", exc_info=True)
+            raise errors.PluginError(
+                "Unable to lock %s", self.conf('config-dir'))
+
+    def more_info(self):
+        """Human-readable string to help the user.
+        Should describe the steps taken and any relevant info to help the user
+        decide which plugin to use.
+        :rtype str:
+        """
+        return (
+            "Configures Postfix to try to authenticate mail servers, use "
+            "installed certificates and disable weak ciphers and protocols.{0}"
+            "Server root: {root}{0}"
+            "Version: {version}".format(
+                os.linesep,
+                root=self.conf('config-dir'),
+                version='.'.join([str(i) for i in self._get_version()]))
+        )
+
+    def _get_version(self):
+        """Return the mail version of Postfix.
+
+        Version is returned as a tuple. (e.g. '2.11.3' is (2, 11, 3))
+
+        :returns: version
+        :rtype: tuple
+
+        :raises .PluginError: Unable to find Postfix version.
+
+        """
+        mail_version = self.postconf.get_default("mail_version")
+        return tuple(int(i) for i in mail_version.split('.'))
+
+    def get_all_names(self):
+        """Returns all names that may be authenticated.
+
+        :rtype: `set` of `str`
+
+        """
+        return certbot_util.get_filtered_names(self.postconf.get(var)
+                   for var in ('mydomain', 'myhostname', 'myorigin',))
+
+
+    def _set_vars(self, var_dict):
+        """Sets all parameters in var_dict to config file.
+        """
+        for param, acceptable in six.iteritems(var_dict):
+            if not util.is_acceptable_value(param, self.postconf.get(param), acceptable):
+                if isinstance(acceptable, tuple):
+                    self.postconf.set(param, acceptable[0], acceptable)
+                else:
+                    self.postconf.set(param, acceptable, acceptable)
+
+    def _confirm_changes(self):
+        """Confirming outstanding updates for configuration parameters.
+
+        :raises errors.PluginError: when user rejects the configuration changes.
+        """
+        updates = self.postconf.get_changes()
+        output_string = "Postfix TLS configuration parameters to update in main.cf:\n"
+        for name, value in six.iteritems(updates):
+            output_string += "{0} = {1}\n".format(name, value)
+        output_string += "Is this okay?\n"
+        if not zope.component.getUtility(interfaces.IDisplay).yesno(output_string, default=False):
```

Comment:
I think we probably want `default=True` here because otherwise when you're running Certbot with -n, --non-interactive, or --noninteractive on the command line we raise the error below.

You can also set `force_interactive=True` if you want. The name of this flag is horrible (I swear it made sense in my head when I wrote it...see #4310), but its main purpose and default value of `False` is to be extra conservative when adding new prompts to existing workflows. Since this is a new plugin, we don't need to be conservative and can always try to prompt when non-interactivity wasn't requested.